syntax = "proto3";
import "sql_options/sql_options.proto";
 
package db;
message tbld_commondata
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 playerid = 2 [(sql).keys = "idx_player"]; //玩家id
    uint32 keytype  = 3;                             //数据类型
    uint32 keyidx   = 4;                             //数据编号
    uint64 data0    = 5;                             //数据1
    uint64 data1    = 6;                             //数据2
    uint64 data2    = 7;                             //数据3
    uint64 data3    = 8;                             //数据4
}

message tbld_cooldown
{

    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 playerid    = 2 [(sql).keys = "idx_player, idx_player_idx"]; //玩家id
    uint32 keytype     = 3 [(sql).keys = "idx_player_idx"];             // CD类型
    uint32 keyidx      = 4 [(sql).keys = "idx_player_idx"];             // CD编号
    uint32 expire_time = 5;                                             // CD结束时间戳
}

message tbld_datacount
{

    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 playerid        = 2 [(sql).keys = "idx_player, idx_player_idx"]; //玩家id
    uint32 keytype         = 3 [(sql).keys = "idx_player_idx"];             // data类型
    uint32 keyidx          = 4 [(sql).keys = "idx_player_idx"];             // data编号
    uint64 data_num        = 5;                                             //累计计数
    uint32 next_reset_time = 6;                                             //下一次重置的时间戳
}

message tbld_global_mail
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 senderid    = 2 [(sql).keys = "idx_sendid"];      //发送者id
    string sendername  = 3 [(sql).size = 255];               //发送者名字
    uint32 create_time = 4 [(sql).keys = "idx_create_time"]; //发送时间戳
    uint64 flag        = 5;                                  //发送者id;
    string subject     = 6 [(sql).size = 255];               //邮件标题
    string content     = 7 [(sql).size = 1024];              //邮件内容
    uint64 filter_type = 8;                                  //玩家过滤条件
    uint64 filter_data = 9;                                  //玩家过滤条件
    bytes  attach_data = 10;                                 //附件数据json
}

message tbld_guild
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 lev  = 2;                    //帮派等级
    string name = 3 [(sql).size = 255]; //帮派名称

    uint64 leader_id    = 4;                      //帮主id
    string leader_name  = 5 [(sql).size = 32];    //帮主名称
    uint64 creator_id   = 6 [(sql).size = 32];    //创建人ID
    string creator_name = 7 [(sql).size = 32];    //创建人名称
    uint32 create_time  = 8;                      //创帮时间
    uint32 del_time     = 9;                      //删除时间
    string slogan       = 10 [(sql).size = 255];  //口号
    string info1        = 11 [(sql).size = 1024]; //信息
    string info2        = 12 [(sql).size = 1024]; //信息
    string info3        = 13 [(sql).size = 1024]; //信息
    uint64 flag         = 14;                     //设置掩码
}

message tbld_item
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 owner_id = 2 [(sql).keys = "idx_owner,idx_owner_pack"]; //所有者id
    uint32 itemtype = 3 [(sql).keys = "idx_itemtype"];             //道具类型编号
    uint32 position = 4 [(sql).keys = "idx_owner_pack"]; //所属包裹类型0:包裹 1:装备槽 100:仓库 101:帮派仓库 102:回购列表 103:邮箱 104:拍卖行

    uint32 grid         = 5;  //所属包裹位置
    uint32 flag         = 6;  //物品掩码
    uint32 pilenum      = 7;  //堆叠数量
    uint32 expire_time  = 8;  //过期时间戳秒
    uint32 dura         = 9;  //当前耐久
    uint32 dura_limit   = 10; //耐久上限
    uint32 addition_lev = 11; //强化等级
    bytes  extra        = 12; //额外信息
}

message tbld_mail
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 ownerid     = 2 [(sql).keys = "idx_ownerid"];     //拥有者id
    uint64 senderid    = 4 [(sql).keys = "idx_sendid"];      //拥有者id
    string sendername  = 5 [(sql).size = 255];               //发送者名字
    uint32 create_time = 6 [(sql).keys = "idx_create_time"]; //发送时间戳
    uint64 flag        = 7;                                  //邮件标志
    string subject     = 8 [(sql).size = 255];               //邮件标题
    string content     = 9 [(sql).size = 1024];              //邮件内容
}

message tbld_mail_attachment
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 mailid      = 2 [(sql).keys = "idx_mailid"]; //邮件id
    uint32 attach_type = 3;                             //附件类型
    uint64 attach_data = 4;                             //数量/具体道具实例ID
}

message tbld_pet
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 ownerid  = 2 [(sql).keys = "idx_ownerid"];  //所有者玩家id
    uint32 pet_type = 3 [(sql).keys = "idx_pet_type"]; //宠物类型
    uint32 hp       = 4;                               //当前血量
    uint64 exp      = 5;                               //当前经验
}

message tbld_player
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint32 worldid    = 2 [(sql).int_size = 16, (sql).keys = "idx_worldid"]; //服务器编号
    uint32 oriworldid = 3 [(sql).int_size = 16];                             //创建角色时的服务器编号
    string openid     = 4 [(sql).size = 255, (sql).keys = "idx_openid"];     //玩家账号
    string name       = 5 [(sql).size = 32, (sql).unique = "idx_name"];      //玩家名字
    uint32 prof       = 6 [(sql).keys = "idx_prof"];                         //玩家职业
    uint32 baselook   = 7;                                                   //玩家基本外观

    uint32 create_time     = 8 [(sql).keys = "idx_create_time"];      //创建时间戳
    uint32 last_logintime  = 9 [(sql).keys = "idx_last_logintime"];   //最后登录时间戳
    uint32 last_logouttime = 10 [(sql).keys = "idx_last_logouttime"]; //最后登出时间戳
    uint32 del_time        = 11 [(sql).keys = "idx_del_time"];        //删除时间戳

    uint32 lev    = 12 [(sql).keys = "idx_lev"]; //玩家等级
    uint32 viplev = 13;                          // vip等级

    uint64 record_sceneid = 14; //玩家当前所在的地图编号
    float  record_x       = 15; //玩家当前所在地图x
    float  record_y       = 16; //玩家当前所在地图y
    float  record_face    = 17; //玩家当前所在地图face
    uint64 home_sceneid   = 18; //玩家回城点对应的地图编号
    float  home_x         = 19; //玩家回城点对应的地图x
    float  home_y         = 20; //玩家回城点对应的地图y
    float  home_face      = 21; //玩家回城点对应的地图face

    uint64 mate_id   = 22;                   //伴侣ID
    string mate_name = 23 [(sql).size = 32]; //伴侣名字
    uint64 teamid    = 24;                   //队伍id
    uint64 guildid   = 25;                   //公会id

    uint64 exp         = 26; //玩家经验
    uint64 money       = 27; //金币
    uint64 money_bind  = 28; //绑定金币
    uint64 gold        = 29; //元宝
    uint64 gold_bind   = 30; //绑定元宝
    uint32 hp          = 31; // hp
    uint32 mp          = 32; // mp
    uint32 fp          = 33; // fp
    uint32 np          = 34; // np
    uint32 pkval       = 35; // pk值
    uint32 honor       = 36; //荣誉值
    uint32 achipoint   = 37; //成就值
    uint32 bag_size    = 38; //背包大小
    uint32 stroge_size = 39; //仓库大小
}

message tbld_player_guildinfo
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 guildid     = 2; // guildid
    uint32 rank        = 3; //帮派职位
    uint32 score       = 4; //帮派贡献
    uint32 total_score = 5; //帮派贡献累计
    uint32 join_time   = 6; //入帮时间
}

message tbld_skill
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 userid    = 2 [(sql).keys = "idx_userid"]; //玩家id
    uint32 skilltype = 3;                             //技能编号
    uint32 lev       = 4;                             //技能等级
}

message tbld_task
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 userid = 2 [(sql).keys = "idx_userid"]; //玩家id
    uint32 taskid = 3 [(sql).keys = "idx_taskid"]; //任务id

    uint32 num0 = 4; //任务计数1
    uint32 num1 = 5; //任务计数2
    uint32 num2 = 6; //任务计数3
    uint32 num3 = 7; //任务计数4

    uint32 accept_userlev = 8;  //接受任务时的等级
    uint32 accept_time    = 9;  //接受任务时间戳
    uint32 finish_time    = 10; //完成任务时间戳
    uint32 expire_time    = 11; //任务过期时间戳
    uint32 daycount       = 12; //本日完成次数
    uint32 daycount_max   = 13; //本日最大可完成次数
    uint32 state          = 14; //任务状态
}

message tbld_achievement
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 userid = 2 [(sql).keys = "idx_userid"]; //玩家id
    uint32 achiid = 3 [(sql).keys = "idx_achiid"]; //成就id
    uint32 take   = 4;                             //成就id
} 

message tbld_status
{
    uint64 id = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    uint64 userid = 2 [(sql).keys = "idx_userid"];   //玩家id
    uint32 typeid = 3 [(sql).keys = "idx_statusid"]; //状态类型编号
    uint32 lev    = 4 [(sql).keys = "idx_statusid"]; //状态等级

    uint32 power = 5; //数据
    uint32 secs  = 6; //持续时间
    uint32 times = 7; //作用次数

    uint32 laststamp = 8;                       //最后一次作用的时间戳
    uint64 casterid  = 9;                       //施加此状态的角色id
    uint32 pause     = 10 [(sql).int_size = 8]; //是否暂停
    uint32 statusid  = 11;                      //状态等级
}

message tbld_systemvar
{
    uint64 keyidx = 1 [(sql).primary_key = "idx_id", (sql).auto_increment = true]; // id

    string name  = 2 [(sql).size = 255];  //系统变量名
    uint64 data0 = 3;                     //数据1
    uint64 data1 = 4;                     //数据2
    uint64 data2 = 5;                     //数据3
    uint64 data3 = 6;                     //数据4
    string str0  = 7 [(sql).size = 255];  //文字1
    string str1  = 8 [(sql).size = 255];  //文字2
    string str2  = 9 [(sql).size = 255];  //文字3
    string str3  = 10 [(sql).size = 255]; //文字4
}
